bentoml/metrics/prometheus.py

```diff
@@ -0,0 +1,109 @@
+from functools import partial
+import logging
+import os
+import shutil
+from typing import Optional, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from multiprocessing.synchronize import Lock
+
+logger = logging.getLogger(__name__)
+
+
+class PrometheusClient:
```

Comment:
What was the motivation behind this refactoring? Seems to introduce more complexity.

Reply:
It improved the code base in several points:
* decoupling the actual implementation `Prometheus` to the interface `metrics_client`. In the future, we may replace the `metrics_client` with other implementations
* you can also find there are no more `setup_prometheus_dir` everywhere.
* We have to keep it in mind: setting the `prometheus_dir` env var before any `import prometheus_client`. This abstraction makes sure it and when we develop the other part, we do not need to care about the Prometheus things anymore.